runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -245,30 +247,37 @@ public long startCheckpoint() {
                 numBytes += cp.getSmrEntriesBytes();
             }
         } else {
-            Iterators.partition(map.keySet().stream()
-                    .map(k -> {
-                        return new SMREntry("put",
-                                new Object[]{keyMutator.apply(k), valueMutator.apply(map.get(k))},
-                                serializer);
-                    }).iterator(), batchSize)
-                    .forEachRemaining(entries -> {
-                        MultiSMREntry smrEntries = new MultiSMREntry();
-                        for (int i = 0; i < ((List) entries).size(); i++) {
-                            smrEntries.addTo((SMREntry) ((List) entries).get(i));
-                        }
-                        CheckpointEntry cp = new CheckpointEntry(CheckpointEntry
-                                .CheckpointEntryType.CONTINUATION,
-                                author, checkpointId, streamId, mdkv, smrEntries);
-                        long pos = sv.append(Collections.singleton(checkpointStreamID), cp, null);
-
-                        postAppendFunc.accept(cp, pos);
-                        continuationAddresses.add(pos);
-
-                        numEntries++;
-                        // CheckpointEntry::serialize() has a side-effect we use
-                        // for an accurate count of serialized bytes of SRMEntries.
-                        numBytes += cp.getSmrEntriesBytes();
-                    });
+
+            TNode[] entries =  ((CMap)map).getEntries();
+
+            int x = 0;
+
+            while (x  < entries.length) {
+                MultiSMREntry smrEntries = new MultiSMREntry();
+                for (int i = x; i < Math.min(x + batchSize, entries.length); i++){
+
+                    smrEntries.addTo(new SMREntry("put",
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SMREntry"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)